diff --git a/modules/cohesion_templates/src/TwigExtension/TwigExtension.php b/modules/cohesion_templates/src/TwigExtension/TwigExtension.php
index 68785eb..28b7450 100644
--- a/modules/cohesion_templates/src/TwigExtension/TwigExtension.php
+++ b/modules/cohesion_templates/src/TwigExtension/TwigExtension.php
@@ -1887,29 +1887,31 @@ class TwigExtension extends AbstractExtension {
             ->load($entity_id);
         }
 
-        $access = $entity->access('view', $this->currentUser, TRUE);
-        if ($entity instanceof EntityInterface && $access) {
-          $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
-          $cache_metadata = CacheableMetadata::createFromObject($access);
-
-          if ($access instanceof AccessResultAllowed) {
-            $build = $this->entityTypeManager->getViewBuilder($entity_type)->view($entity, $view_mode, $language);
-          }
+        if ($entity !== NULL) {
+          $access = $entity->access('view', $this->currentUser, TRUE);
+          if ($entity instanceof EntityInterface && $access) {
+            $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+            $cache_metadata = CacheableMetadata::createFromObject($access);
+
+            if ($access instanceof AccessResultAllowed) {
+              $build = $this->entityTypeManager->getViewBuilder($entity_type)->view($entity, $view_mode, $language);
+            }
 
-          else {
-            $build = [
-              "#cache" => [
-                "contexts" => $entity->getCacheContexts(),
-                "tags" => $entity->getCacheTags(),
-                "max-age" => $entity->getCacheMaxAge(),
-              ],
-            ];
-          }
+            else {
+              $build = [
+                "#cache" => [
+                  "contexts" => $entity->getCacheContexts(),
+                  "tags" => $entity->getCacheTags(),
+                  "max-age" => $entity->getCacheMaxAge(),
+                ],
+              ];
+            }
 
-          $build['#cache']['contexts'] = array_merge($build['#cache']['contexts'], $cache_metadata->getCacheContexts());
-          $build['#cache']['tags'] = array_merge($build['#cache']['tags'], $cache_metadata->getCacheTags());
+            $build['#cache']['contexts'] = array_merge($build['#cache']['contexts'], $cache_metadata->getCacheContexts());
+            $build['#cache']['tags'] = array_merge($build['#cache']['tags'], $cache_metadata->getCacheTags());
 
-          return $build;
+            return $build;
+          }
         }
       }
 
