diff --git a/src/Form/openaiImagesGenerator.php b/src/Form/openaiImagesGenerator.php
index dbfe7f4..c91e0c6 100644
--- a/src/Form/openaiImagesGenerator.php
+++ b/src/Form/openaiImagesGenerator.php
@@ -298,7 +298,7 @@ class openaiImagesGenerator extends FormBase
       // Create a media entity
       $media = Media::create([
         'bundle' => 'image',
-        'field_media_image' => [
+        'image' => [
           'target_id' => $file->id(),
         ],
       ]);
diff --git a/src/Form/openaiImagesVariations.php b/src/Form/openaiImagesVariations.php
index 5beff10..c43c085 100644
--- a/src/Form/openaiImagesVariations.php
+++ b/src/Form/openaiImagesVariations.php
@@ -335,7 +335,7 @@ class openaiImagesVariations extends FormBase
       // Create a media entity
       $media = Media::create([
         'bundle' => 'image',
-        'field_media_image' => [
+        'image' => [
           'target_id' => $file->id(),
         ],
       ]);
