diff --git a/src/Form/openaiImagesVariations.php b/src/Form/openaiImagesVariations.php
index 5beff10..2ee3850 100644
--- a/src/Form/openaiImagesVariations.php
+++ b/src/Form/openaiImagesVariations.php
@@ -257,7 +257,7 @@ class openaiImagesVariations extends FormBase
     $api_key = $this->config('openai_images.settings')->get('api_key');
 
     // Read the image file and convert it to content.
-    $file_url = Url::fromUri(file_create_url($image_file))->toString();
+    $file_url = \Drupal::service('file_system')->realpath($image_file);
     $image_data = file_get_contents($file_url);
 
     // Prepare the form data
