diff --git a/src/Form/MonsidoSettingsForm.php b/src/Form/MonsidoSettingsForm.php
index 3595c64..d133fed 100644
--- a/src/Form/MonsidoSettingsForm.php
+++ b/src/Form/MonsidoSettingsForm.php
@@ -74,7 +74,7 @@ class MonsidoSettingsForm extends ConfigFormBase {
       '#title' => $this->t('Script from Monsido.com'),
       '#default_value' => $config->get('monsido_script'),
       '#description' => $this->t('Paste in the generated script from the Monsido web app. Please provide the entire script as generated.'),
-      '#required' => TRUE,
+      '#required' => FALSE,
       '#states' => [
         'visible' => [
           ':input[name=enable]' => ['checked' => TRUE],
